### PR TITLE
regression: 4005: add nist aes-gcm vectors

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Adds nist AES-GCM test vectors to regression case 4005 with
CFG_GCM_NIST_VECTORS=y. Only the first test in each group is
used if CFG_GCM_NIST_VECTORS_LIMITED=y.

With CFG_GCM_NIST_VECTORS=y the size of the xtest binary grows with more
than 10 MiB, unless CFG_GCM_NIST_VECTORS_LIMITED=y is given, then only
by ~1 MiB.

Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

File: host/xtest/regression_4000.c [hunk 1]
```diff
@@ -2654,8 +2654,8 @@ static const struct xtest_ae_case ae_cases[] = {
 	XTEST_AE_CASE_AES_GCM(vect1, 0, 0, NULL_ARRAY, NULL_ARRAY, NULL_ARRAY),
 	XTEST_AE_CASE_AES_GCM(vect2, 0, 9, NULL_ARRAY, ARRAY, ARRAY),
 	XTEST_AE_CASE_AES_GCM(vect3, 0, 9, NULL_ARRAY, ARRAY, ARRAY),
-	XTEST_AE_CASE_AES_GCM(vect3, 0, 0x20, NULL_ARRAY, ARRAY, ARRAY),
-	XTEST_AE_CASE_AES_GCM(vect4, 5, 9, ARRAY, ARRAY, ARRAY),
+	XTEST_AE_CASE_AES_GCM(vect3, 0, 0x1F, NULL_ARRAY, ARRAY, ARRAY),
+	XTEST_AE_CASE_AES_GCM(vect4, 5, 0x20, ARRAY, ARRAY, ARRAY),
```
Comment (jforissier) [inline host/xtest/regression_4000.c]: Why?

File: scripts/rsp_to_gcm_test.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
```
Comment (jforissier) [inline scripts/rsp_to_gcm_test.py]: - Does it run with Python 2 & 3?
- Better make this script pass `pycodestyle`/`pep8`

File: host/xtest/regression_4000.c (hunk 1 above)
Comment (jenswi-linaro) [reply host/xtest/regression_4000.c]: It increases test coverage, it should be in a separate commit though. I'll fix.

File: scripts/rsp_to_gcm_test.py (hunk 2 above)
Comment (jenswi-linaro) [reply scripts/rsp_to_gcm_test.py]: Yes it seems to work with both Python 2 and 3.
I'll fix the style stuff.

**Comment by jenswi-linaro** [comment]
Updated and rebased

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey960)`

That's a **lot** of test cases indeed :open_mouth: 

```
root@HiKey960:/ time xtest 4005
...
o regression_4005.25290 AE case 25289 algo 0x40000810 line 65897
  regression_4005.25290 OK
  regression_4005 OK
...
real	3m 22.96s
user	0m 2.22s
sys	1m 2.48s
...

**Comment by jenswi-linaro** [comment]
Tags applied

**Pull request merged**